### ISSUE TITLE
adjust DeepCWrapper and DeepCMWrapper

### ISSUE THREAD

**Issue by falkhofmann** [charlesangus/DeepC#56]
I would like to do some minor adjustment on the wrapper classes.

mainly i want to inject a vector reference into the wrappedPer... methods.
I would need those to apply matrix calculations and colorspace shifts which using all channels of a set.

I have tested and adjusted all other nodes on a local branch building a DeepColorMatrix node. Are there any objections to do such a thing @charlesangus ?

personally i would prefer this approach then having the get the brothers on each perChannel call.

basically it would look like this with the additional sampleColor.

```
  virtual void wrappedPerSample(
      Box::iterator it,
      size_t sampleNo,
      float alpha,
      DeepPixel deepInPixel,
      float &perSampleData,
      Vector3 &sampleColor
      );

  virtual void wrappedPerChannel(
      const float inputVal,
      float perSampleData,
      Channel z,
      float& outData,
      Vector3& sampleColor
      );
```

**Comment by charlesangus** [comment]
Not in principle, but is there a performance implication for the "simple" nodes which don't require it?

Maybe test a 300f render (animated noise with a few samples per pixel or something) using DeepCGrade under both options?

I wouldn't think there would be much but better to be sure.

**Comment by falkhofmann** [comment]
sure its definitely better to check upfront.

i did as you suggested a 300f sequence created inside nuke to avoid I/O side effects.
Basic setup on 1080p:
DeepCConstant > DeepCPNoise (animated) > 3 DeepCGrades with masks > DeepToImage > Write


both versions are fairly close.
"current": 12.23 mins
"adjusted": 12.31 mins

since this is not a proper benchmark, its hard to say if this actually coming from the adjustments or simply "noise" on my machine.
However, given the setup and assuming its really caused by the adjustments, 8 seconds on volumetric deep grading with 10 samples over this very range and nodes, it should be ok.

therefore i would go ahead and adjust the wrapper and nodes accordingly.
